engine/robot.py

```diff
@@ -152,10 +186,9 @@ def move_forward(self, dist):
             self.truthpose = np.append(
                 self.truthpose, np.transpose(self.state), 0)
 
-    def move_to_target_node(self, target, allowed_dist_error, database):
+    def move_to_target_node(self, target, allowed_dist_error, database, mc):
```

Comment:
instead of adding a new `mc` parameter here, can we use the motor_controller attribute of the robot class?